packages/automerge-repo/src/DocHandle.ts

```diff
@@ -89,70 +112,111 @@ export class DocHandle<T> //
               if (docChanged) this.emit("change", { handle: this })
               return { doc: newDoc }
             }),
-
-            failTimeout: _ => {},
           },
         }
       )
     )
       .onTransition(({ value: state }, { type: event }) =>
-        this.#log(`${event} → ${state}`, this.doc)
+        this.#log(`${event} → ${state}`, this.#doc)
       )
       .start()
 
     this.#machine.send(isNew ? CREATE : FIND)
   }
 
-  // PUBLIC
+  // PRIVATE
 
-  get doc() {
-    return this.#machine?.getSnapshot().context.doc || ({} as A.Doc<T>)
+  /** Returns the current document */
+  get #doc() {
```

Comment:
this accessor shouldn't be used externally — consumers should always go through `value()`